doc/specs/#4999 - Improved keyboard handling in Conpty.md

```diff
@@ -0,0 +1,514 @@
+```
+^[[16;42;;1;16_
+^[[65;30;65;1;16_
+^[[16;42_
+^[[65;30;97_
+```
+
+* Or even easier, just <kbd>a</kbd>
+```
+^[[65;30;97;1_
+^[[65;30;97_
+```
+
+### Scenarios
+
+#### User is typing into WSL from the Windows Terminal
+
+`WT -> conpty[1] -> wsl`
+
+* Conpty[1] will ask for `win32-input-mode` from the Windows Terminal when
+  conpty[1] first boots up. Conpty will _always_ as for win32-input-mode -
+  Terminals that _don't_ support this mode will ignore this sequence on startup.
+* When the user types keys in Windows Terminal, WT will translate them into
+  win32 sequences and send them to conpty[1]
+* Conpty[1] will translate those win32 sequences into `INPUT_RECORD`s.
+  - When those `INPUT_RECORD`s are written to the input buffer, they'll be
+    converted into VT sequences corresponding to whatever input mode the linux
+    app is in.
+* When WSL reads the input, it'll read (using `ReadConsoleInput`) a stream of
+  `INPUT_RECORD`s that contain only character information, which it will then
+  pass to the linux application.
+  - This is how `wsl.exe` behaves today, before this change.
+
+#### User is typing into `cmd.exe` running in WSL interop
+
+`WT -> conpty[1] -> wsl -> conpty[2] -> cmd.exe`
+
+(presuming you start from the previous scenario, and launch `cmd.exe` inside wsl)
+
+* Conpty[2] will ask for `win32-input-mode` from conpty[1] when conpty[2] first
+  boots up.
+    - As conpty[1] is just a conhost that knows how to handle
+      `win32-input-mode`, it will switch its own VT input handling into
+      `win32-input-mode`
+* When the user types keys in Windows Terminal, WT will translate them into
+  win32 sequences and send them to conpty[1]
+* Conpty[1] will translate those win32 sequences into `INPUT_RECORD`s. When
+  conpty[1] writes these to its buffer, it will translate the `INPUT_RECORD`s
+  into VT sequences for the `win32-input-mode`. This is because it believes the
+  client (in this case, the conpty[2] running attached to `wsl`) wants
+  `win32-input-mode`.
+* When WSL reads the input, it'll read (using `ReadConsoleInput`) a stream of
+  `INPUT_RECORD`s that contain only character information, which it will then
+  use to pass a stream of characters to conpty[2].
+* Conpty[2] will get those sequences, and will translate those win32 sequences
+  into `INPUT_RECORD`s
+* When `cmd.exe` reads the input, they'll receive the full `INPUT_RECORD`s
+  they're expecting
+
+
+## UI/UX Design
+
+This is not a user-facing feature.
+
+## Capabilities
+
+### Accessibility
+
+_(no change expected)_
+
+### Security
+
+_(no change expected)_
+
+### Reliability
+
+_(no change expected)_
+
+### Compatibility
+
+This isn't expected to break any existing scenarios. The information that we're
+passing to conpty from the Terminal should strictly have _more_ information in
+them than they used to. Conhost was already capable of translating
+`INPUT_RECORD`s back into VT sequences, so this should work the same as before.

```

Comment:
**must validate**

* [ ] <kbd>Ctrl+H</kbd>
* [ ] <kbd>Ctrl+C</kbd> in the specific instance of #1894
* [ ] **anything requiring <kbd>AltGr</kbd>**
* [ ] Anything as above, but with the <kbd>Ctrl+Alt</kbd> fallback toggle from #6211